sql-odbc/src/sqlodbc/opensearch_connection.cpp

```diff
@@ -73,6 +75,26 @@ char CC_connect(ConnectionClass *self) {
     return 1;
 }
 
+std::string generateValidServerUrl(ConnectionClass *self) {
+    std::string valid_server_url = "";
+    if(!strlen(self->connInfo.server)) {
```

Comment:
You could use string_view to wrap `self->connInfo.server` into a cstring interface.